app/models/az321_contribution.rb

```diff
@@ -16,5 +16,20 @@
 #  index_az321_contributions_on_state_file_az_intake_id  (state_file_az_intake_id)
 #
 class Az321Contribution < ApplicationRecord
+  attr_accessor :made_contributions
+  date_accessor :date_of_contribution
+
   belongs_to :state_file_az_intake
+
+  # Virtual attribute, not in database. Only checked when created via form interface.
+  validates :made_contributions, presence: true, on: :form_create
+
+  validates :charity_name, presence: true
+  validates :charity_code, presence: true
```

Comment:
ok i think i messed up by making this a string because looking at the list of QCOs it seems like it has to be a 5 digit number https://azdor.gov/sites/default/files/2023-05/CREDITS_2023_qco.pdf. we could either change the db or just add the validation step?

Reply:
Sure, could do both.

Reply:
ok! just to clarify: did you mean you'll do that as part of this PR or later? i'd be fine with just the validation tbh and it might even be good to ask tiffany or program if that is the correct validation

Reply:
I was planning on doing this as part of the PR but I can make a note to talk to Tiffany while I work on other changes and if she doesn't get back to me before I'm finished I'll punt and make a new ticket to follow up.

Reply:
Tiffany got back to me and updated the story

![image](https://github.com/user-attachments/assets/f8809bf3-26f6-439d-a339-746437d7a717)

Not sure atm if I'm going to do integer w/ inclusion or something with string. If the id had leading zeros it'd be easy to do a string but if it's always between these values it might make sense to migrate the column. Will think on it.